src/dnx.coreclr.unix/dnx.coreclr.cpp

```diff
@@ -0,0 +1,227 @@
+#include "stdafx.h"
+#include "tpa.h"
+
+// Windows types used by the ExecuteAssembly function
+typedef int32_t HRESULT;
+typedef const char* LPCSTR;
+typedef unsigned int DWORD;
+#define S_OK 0;
+#define E_FAIL -1;
+
+typedef struct CALL_APPLICTION_MAIN_DATA
+{
+    char* applicationBase;
+    char* runtimeDirectory;
+    int argc;
+    char** argv;
+    int exitcode;
+} *PCALL_APPLICATION_MAIN_DATA;
+
+// Prototype of the ExecuteAssembly function from the libcoreclr.do
+typedef HRESULT (*ExecuteAssemblyFunction)(
+                    LPCSTR exePath,
+                    LPCSTR coreClrPath,
+                    LPCSTR appDomainFriendlyName,
+                    int propertyCount,
+                    LPCSTR* propertyKeys,
+                    LPCSTR* propertyValues,
+                    int argc,
+                    LPCSTR* argv,
+                    LPCSTR managedAssemblyPath,
+                    LPCSTR entryPointAssemblyName,
+                    LPCSTR entryPointTypeName,
+                    LPCSTR entryPointMethodsName,
+                    DWORD* exitCode);
+
+bool GetTrustedPlatformAssembliesList(const std::string& szDirectory, bool bNative, std::string& trustedPlatformAssemblies)
```

Comment:
seems like we should skip the Hungarian for szDirectory, at least. Maybe bNative too since it's a C++ bool.